fix(MIGRATING.md): enhance migration guide to clarify migration steps from v1 to v2 (#3575)
DataDog/dd-trace-go: MIGRATING.md

Our [godoc page](https://pkg.go.dev/github.com/DataDog/dd-trace-go/v2/ddtrace) should be helpful during this process. We also have the [official documentation](https://docs.datadoghq.com/tracing/setup/go/), which contains a couple of examples.

> Please follow this guide for migrating from v1 to v2 and creating new services using v2 directly. This document will further outline some _before_ and _after_ examples.

Reviewer: Adding a link would be nice.

Author: @kakkoyun It's referring to the guide itself. Where should we link?

import "github.com/DataDog/dd-trace-go/v2/ddtrace/tracer"
```

It is important to run `go mod tidy` after changing any import.

It is also important to note that when using our contrib libraries, import URLs may be impacted differently. This will be covered in the section below [Independent contrib packages](#independent-contrib-packages).

### Gradual code repair

For customers that have large codebases and that are unable to immediately upgrade services to the v2 tracer, we are offering a special transitional version of the v1 tracer, the [v1 transitional version](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.74.0) which is essentially a wrapper that preserves the API from v1 but uses v2 under the hood.

> You should import only one version of the tracer for any given service (with the exception of the v1 transitional version), but your organization can import different versions of the tracer for different services as you perform your migration. Staggering the upgrade per service can minimize downtime and help ensure stability throughout your transition.

Reviewer 2: ```suggestion
It’s safe for different services within your organization to use different tracer versions during the transition, some can use v1 while others adopt v2. Staggering the upgrade per service can minimize downtime and help ensure stability throughout your transition.
```

I don't think we need the first part as we explain just after that they can use `v1.74.0` alongside v2 in the same service.

> The transitional version can coexist with v2 in the same service, which enables you to migrate services gradually. Using this v1 transitional version doesn’t require any code changes, but it also won’t support any new features of the new v2 tracer.

Reviewer 2: ```suggestion
The transitional version is the only v1 version that allows both v1 and v2 to be imported in the same service, which enables you to migrate a service gradually. Using this v1 transitional version doesn’t require any code changes, but it also won’t support any new features introduced in v2.
```

	tracer.SpanType(ext.SpanTypeWeb),
	tracer.Tag("key", "value"),
)

> finishCfg := tracer.NewFinishConfig(

Reviewer: ref: https://github.com/DataDog/dd-trace-go/pull/3574
